Lib/test/test_float.py

```diff
@@ -1468,6 +1468,20 @@ def test_from_hex(self):
         self.identical(fromHex('0X1.0000000000001fp0'), 1.0+2*EPS)
         self.identical(fromHex('0x1.00000000000020p0'), 1.0+2*EPS)
 
+        # Regression test for a corner-case bug reported in b.p.o. 44954
+        self.identical(fromHex('0x.8p-1074'), 0.0)
+        self.identical(fromHex('0x.80p-1074'), 0.0)
+        self.identical(fromHex('0x.81p-1074'), TINY)
```

Comment:
I tried to find what is the minimal example for non-zero result, but seems it works for arbitrary number of zeroes between 8 and 1: `0x.800000000000000000000...0000000000000000001p-1074`.

Reply:
Thanks, that's reassuring, and as it should be: `0x.8p-1074` is the exact threshold for rounding, so if it's larger than `0x.8p-1074` by even the tiniest amount, it should round up.